topdown/encoding.go

```diff
@@ -35,6 +35,39 @@ func builtinJSONMarshal(_ BuiltinContext, operands []*ast.Term, iter func(*ast.T
 	return iter(ast.StringTerm(string(bs)))
 }
 
+func builtinJSONMarshalIndent(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+
+	asJSON, err := ast.JSON(operands[0].Value)
+	if err != nil {
+		return err
+	}
+
+	var indentWith string
+
+	switch operands[1].Value.(type) {
+	case ast.String:
+		indentOp, err := builtins.StringOperand(operands[1].Value, 2)
+		if err != nil {
+			return err
+		}
+
+		indentWith = string(indentOp)
+
+	case ast.Null:
+		indentWith = "\t"
+
+	default:
+		return builtins.NewOperandTypeErr(2, operands[1].Value, "string", "null")
+	}
+
+	bs, err := json.MarshalIndent(asJSON, "", indentWith)
```

Comment:
Recalling your original use case (generating JSON for markdown output), it might be helpful for other users in similar cases to be able to specify the prefix.

As I see it, this is the flexible version of `json.marshal` and it'd be preferred to have all the options available. Prefixing all lines with 4 spaces, might allow users to generate indented code blocks etc.

Reply:
I agree that this would be easiest to add now (before the built-in has shipped to customers) so there's no chance of dependencies on a prefix-free function interface.

One of the reasons I shied away from adding the prefix argument was that golang's [MarshalIndent](https://pkg.go.dev/encoding/json#MarshalIndent) (more specifically [Indent](https://pkg.go.dev/encoding/json#Indent)) has a strange "feature" in that the first line is emitted sans-prefix. 

This isn't even included in the documentation for MarshalIndent proper, and I wanted to avoid violating the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

I see two options for adding a prefix parameter:
* Do not abstract over Golang's `MarshalIndent()` prefix behavior, and call it out in the rego docs.  
This seems important enough to warrant a highlighted warning (or similar) with contextual links to the Golang docs, and I wasn't sure how to structure the function definition (or manually edit the docs) such that the docs were formatted appropriately.
* Add in a copy of the prefix string before the first line of the emitted json payload. This naturally (and intentionally) deviates from Go's implementation, but I'd posit that knowledge of this implementation (or lack thereof) shouldn't be necessary for using OPA.

Reply:
I've added a prefix param that aligns with Option 2 described above. Looking forward to your thoughts on this.